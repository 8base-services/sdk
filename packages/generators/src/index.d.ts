import { TableSchema } from '@8base/utils';

export const REPO_BRANCH_NAME: string;

export type GeneratorsCommonConfig = {
  deep: number,
  withMeta?: boolean,
};

export type GeneratorsConfig = {
  includeColumns?: string[],
} & GeneratorsCommonConfig;

export type GeneratorsData = {
  tablesList: TableSchema,
  tableName: string,
  screenName?: string,
};

export type ScreenTable = {
  id?: string,
  tableName: string,
  screenName: string,
  routeUrl?: string,
  tableFields?: string[],
  formFields?: string[],
};


export function replaceInitialApp(fsObject: Object, constants: {
  endpoint: string,
  authClientId: string,
  authDomain: string,
  appName: string,
}): Object;

export function generateScreen(data: {
  tablesList: TableSchema[],
  screen: ScreenTable,
  rootFile?: string,
}, config: GeneratorsCommonConfig): Object;
