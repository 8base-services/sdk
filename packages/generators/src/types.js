// @flow
import type { TableSchema } from '@8base/utils';

export type GeneratorsConfig = {
  deep: number,
  includeColumns?: string[],
};

export type GeneratorsData = {
  tablesList: TableSchema,
  tableName: string,
  screenName?: string,
};

export type ScreenTable = {
  id: string,
  tableName: string,
  screenName: string,
  routeUrl: string,
  tableFields?: string[],
  formFields?: string[],
};
