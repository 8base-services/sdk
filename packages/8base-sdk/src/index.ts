export * from '@8base/apollo-client';
export * from '@8base/apollo-links';
export * from '@8base/auth';
export * from '@8base/utils';
export * from '@8base/validate';
