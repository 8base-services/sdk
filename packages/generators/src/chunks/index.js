// @flow
import ejs from 'ejs';

// $FlowIgnore
import fieldsInputsList from './fieldsInputsList.js.ejs';
// $FlowIgnore
import fieldsQueries from './fieldsQueries.js.ejs';
// $FlowIgnore
import fieldsFileInput from './fieldsFileInput.js.ejs';

export const chunks = {
  fieldsInputsList: (data?: Object) => ejs.render(fieldsInputsList, data),
  fieldsQueries: (data?: Object) => ejs.render(fieldsQueries, data),
  fieldsFileInput: (data?: Object) => ejs.render(fieldsFileInput, data),
};

