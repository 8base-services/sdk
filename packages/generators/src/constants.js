// @flow

/** Name of thi branch in repo https://github.com/8base/react-app-starter */
export const REPO_BRANCH_NAME = 'v1.1.0';
