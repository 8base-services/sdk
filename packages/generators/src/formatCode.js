// @flow
import prettier from 'prettier/standalone';
import prettierBabylon from 'prettier/parser-babylon';

export const formatCode = (code: string) => prettier.format(code, {
  printWidth: 120,
  singleQuote: true,
  trailingComma: 'es5',
  parser: 'babel',
  plugins: [prettierBabylon],
});
