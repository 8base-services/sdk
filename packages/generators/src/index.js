// @flow

export { generateCreateForm } from './generateCreateForm/createForm';
export { generateUpdateForm } from './generateUpdateForm/updateForm';
export { generateDeleteForm } from './generateDeleteForm/deleteForm';
export { generateTable } from './generateTable/table';

