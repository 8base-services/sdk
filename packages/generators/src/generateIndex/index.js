// @flow

import ejs from 'ejs';
import pluralize from 'pluralize';
import changeCase from 'change-case';
import { formatCode } from '../formatCode';

// $FlowIgnore
import index from './index.js.ejs';

export const generateIndex = ({ tableName, screenName }: *) => {
  const entityName = pluralize.singular(tableName);

  const tableGenerated = ejs.render(index, {
    changeCase,
    pluralize,
    tableName,
    entityName,
    screenName: screenName || entityName,
  });

  return formatCode(tableGenerated);
};

