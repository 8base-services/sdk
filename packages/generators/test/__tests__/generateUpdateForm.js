// @flow

import { generateUpdateForm } from '../../src';
import { tablesSchema } from '../__fixtures__/tablesSchema';


it('should generate update form by the table name', () => {
  const generatedTable = generateUpdateForm(tablesSchema, 'Properties');

  expect(generatedTable).toMatchSnapshot();
});
