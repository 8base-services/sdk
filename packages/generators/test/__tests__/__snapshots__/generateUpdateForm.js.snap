// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should generate update form by the table name 1`] = `
"import React from 'react';
import * as R from 'ramda';
import { Form, Field } from '@8base/forms';
import { Dialog, Grid, Button, InputField, SelectField, ModalContext } from '@8base/boost';
import { graphql, Query } from 'react-apollo';


const PROPERTY_UPDATE_MUTATION = gql\`
  mutation PropertyUpdate($data: PropertyUpdateInput!) {
    propertyUpdate(data: $data) {
      id
    }
  }
\`;

const LISTINGS_LIST_QUERY = gql\`
  query ListingsListList {
    listingsList {
      items {
        id
        _description
      }
    }
  }
\`;



const getRelationOptions = (items = []) =>
  items.map(item => ({ value: item.id, label: item._description }));


const FileInputField = ({ input, meta, maxFiles, label, ...rest }) => (
  <Form.Field label={label} input={input} meta={meta}>
    <FileInput onChange={input.onChange} value={input.value} maxFiles={maxFiles} public={rest.public}>
      {({ pick, value }) => (
        <Row stretch alignItems=\\"center\\">
          <Button type=\\"button\\" onClick={pick} stretch color=\\"neutral\\">
            Choose Files
          </Button>
          <Text size=\\"sm\\" style={{ whiteSpace: 'nowrap' }}>
            {value ? (Array.isArray(value) ? \`\${value.length} files selected\` : value.filename) : 'No files selected'}
          </Text>
        </Row>
      )}
    </FileInput>
  </Form.Field>
);


const ehnhancer = graphql( PROPERTY_UPDATE_MUTATION, {
  name: 'propertyUpdate',
  options: {
    refetchQueries: ['propertiesList'],
    context: {
      ['TOAST_SUCCESS_MESSAGE']: 'Property successfuly updated',
    },
  },
});


const PropertyEditDialog = (
  class ListingEditDialog extends React.PureComponent {
    static contextType = ModalContext;

    createOnSubmit = R.memoizeWith(R.identity, id => async data => {
      await this.props.listingUpdate({ variables: { data: { ...data, id } } });

      this.context.closeModal('PROPERTY_EDIT_DIALOG_ID');
    });

    onClose = () => {
      this.context.closeModal('PROPERTY_EDIT_DIALOG_ID');
    };

    renderFormContent = ({ handleSubmit, invalid, submitting, pristine }) => (
      <form onSubmit={handleSubmit}>
        <Dialog.Header title=\\"Edit Property\\" onClose={this.onClose} />
        <Dialog.Body scrollable>
          <Grid.Layout gap=\\"sm\\" stretch>
                    
          <Grid.Box>
            <Field name=\\"pictures\\" label=\\"Pictures\\" type=\\"text\\" component={FileInputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"bedrooms\\" label=\\"Bedrooms\\" type=\\"number\\" component={InputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"title\\" label=\\"title\\" type=\\"text\\" component={InputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"description\\" label=\\"Description\\" type=\\"text\\" component={InputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"sqFootage\\" label=\\"Sq Footage\\" type=\\"number\\" component={InputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"bathrooms\\" label=\\"Bathrooms\\" type=\\"number\\" component={InputField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"garage\\" label=\\"Garage\\" component={CheckboxField} />
          </Grid.Box>
        
          <Grid.Box>
            <Field name=\\"pool\\" label=\\"Pool\\" component={CheckboxField} />
          </Grid.Box>
        
          <Grid.Box>
            <Query query={LISTINGS_LIST_QUERY}>
                {({ data, loading }) => (
                  <Field
                    name=\\"listings\\" 
                    label=\\"Listings\\"
                    placeholder=\\"Select a listing\\"
                    component={SelectField}
                    loading={loading}
                    options={
                      loading
                        ? []
                        : getRelationOptions(data.listingsList.items)
                    }
                    stretch
                  />
                )}
              </Query>
          </Grid.Box>
        

          </Grid.Layout>
        </Dialog.Body>
        <Dialog.Footer>
          <Button color=\\"neutral\\" variant=\\"outlined\\" disabled={submitting} onClick={this.onClose}>
            Cancel
          </Button>
          <Button color=\\"primary\\" type=\\"submit\\" disabled={pristine || invalid} loading={submitting}>
            Update Listing
          </Button>
        </Dialog.Footer>
      </form>
    );

    renderForm = ({ args }) => {
      return (
        <Form
          type=\\"UPDATE\\"
          tableSchemaName=\\"Properties\\"
          onSubmit={this.createOnSubmit(args.initialValues.id)}
          initialValues={args.initialValues}
        >
          {this.renderFormContent}
        </Form>
      );
    };

    render() {
      return (
        <Dialog id={'PROPERTY_EDIT_DIALOG_ID'} size=\\"sm\\">
          {this.renderForm}
        </Dialog>
      );
    }
  }
);

export { PropertyEditDialog };
"
`;
